app/migrations/Version20231205094436.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Mautic\CoreBundle\Doctrine\PreUpAssertionMigration;
+use Mautic\LeadBundle\Entity\ListLead;
+
+final class Version20231205094436 extends PreUpAssertionMigration
+{
+    protected function preUpAssertions(): void
+    {
+        $this->skipAssertion(function (Schema $schema) {
+            return $schema->getTable($this->getTableName())->hasIndex($this->getIndexName());
+        }, sprintf('Index %s already exists', $this->getIndexName()));
+    }
+
+    public function up(Schema $schema): void
+    {
+        $this->addSql(sprintf(
+            'ALTER TABLE %s ADD INDEX %s (lead_id, leadlist_id, manually_removed)',
+            $this->getTableName(),
+            $this->getIndexName()
+        ));
+    }
+
+    public function down(Schema $schema): void
+    {
+        $this->addSql('ALTER TABLE '.$this->getTableName().' DROP INDEX '.$this->getIndexName());
+    }
+
+    private function getTableName(): string
+    {
+        return ListLead::TABLE_NAME;
+    }
+
+    private function getIndexName(): string
+    {
+        return 'lead_id_lists_id_removed';
```

Comment:
@aarohiprasad can you check why the prefix isn't here? The original PR has it:
```php
return "{$this->prefix}lead_id_lists_id_removed";
```
Why was it removed here?

Reply:
Whoops!! I removed it to check something out for myself. I should've been more mindful. I have added it back!

Reply:
Ok. Such changes are very hard to spot and can cause serious headaches after the release. Please be more careful.